API/src/main/java/net/juligames/core/api/message/MessageApi.java

```diff
@@ -0,0 +1,188 @@
+
+    @ApiStatus.Internal
+    @ApiStatus.Experimental
+    <R>
+    R callReplacementExtension(ExtensionCallback<R, ReplacementDAO, RuntimeException> extensionCallback);
+
+    @ApiStatus.Internal
+    @ApiStatus.Experimental
+    <R>
+    R callReplacementTypeExtension(ExtensionCallback<R, ReplacementTypeDAO, RuntimeException> extensionCallback);
+
+    //get
+    Message getMessage(String messageKey, Locale locale);
+
+    Message getMessage(String messageKey, Locale locale, String... replacements);
+
+    Message getMessage(String messageKey, String locale);
+
+    Message getMessage(String messageKey, String locale, String... replacements);
+
+    Message getMessage(String messageKey, DBLocale dbLocale);
+
+    Message getMessage(String messageKey, DBLocale dbLocale, String... replacements);
+
+    Collection<? extends Message> getMessage(String messageKey);
+
+    Collection<? extends Message> getMessage(String messageKey, String... replacements);
+
+    Collection<? extends Message> getAllFromLocale(Locale locale);
+
+    Collection<? extends Message> getAllFromLocale(Locale locale, String... replacements);
+
+    Collection<? extends Message> getAllFromLocale(String locale);
+
+    Collection<? extends Message> getAllFromLocale(String locale, String... replacements);
+
+    Collection<? extends Message> getAllFromLocale(DBLocale dbLocale);
+
+    Collection<? extends Message> getAllFromLocale(DBLocale dbLocale, String... replacements);
+
+    @ApiStatus.Experimental
+    Collection<? extends Message> getAll();
+
+    @ApiStatus.Experimental
+    Collection<? extends Message> getAll(String... replacements);
+
+    @ApiStatus.Internal
+    Stream<? extends DBMessage> streamData();
+
+    Collection<? extends DBReplacement> getReplacers();
+
+    //register
+
+    @Deprecated
+    void registerMessage(String messageKey);
+
+    void registerMessage(String messageKey, String defaultMiniMessage);
+
+    boolean hasMessage(String messageKey);
+
+    boolean hasMessage(String messageKey, String locale);
+
+    boolean hasMessage(String messageKey, Locale locale);
+
+    boolean hasMessage(String messageKey, DBLocale locale);
+
+    MessagePostScript sendMessage(String messageKey, MessageRecipient messageRecipient);
+
+    MessagePostScript sendMessage(String messageKey, MessageRecipient messageRecipient, Locale overrideLocale);
+
+    MessagePostScript sendMessage(String messageKey, MessageRecipient messageRecipient, String overrideLocale);
+
+    MessagePostScript sendMessage(String messageKey, MessageRecipient messageRecipient, DBLocale overrideLocale);
+
+    MultiMessagePostScript broadcastMessage(String messageKey, Locale defaultLocale);
+
+    MultiMessagePostScript broadcastMessage(String messageKey, String defaultLocale);
+
+    MultiMessagePostScript broadcastMessage(String messageKey, DBLocale defaultLocale);
+
+    Collection<? extends MessagePostScript> broadcastMessage(String messageKey);
```

Comment:
```sugge```suggestion
    Collection<? extends MessagePostScript> broadcastMessage(String messageKey);
```stion
    Collection<? extends MessagePostScript> broadcastMessage(String messageKey);
```

Reply:
Please explain your request more precise

Reply:
resolved